src/dataframe/dataframe.jl

```diff
@@ -1508,3 +1508,63 @@ end
 function permutecols!(df::DataFrame, p::AbstractVector{Symbol})
     permutecols!(df, index(df)[p])
 end
+
+##############################################################################
+##
+## Expand & Complete dataframes
+##
+##############################################################################
+
+
+function expanddf(df::AbstractDataFrame, indexcols::Array{Symbol,1})
+    colnames = names(df)
+    # Check to make sure the symbols in indexcols are in the df
+    for i in indexcols
+        if i ∉ colnames
+            throw(ArgumentError(":$i is not in dataframe"))
+        end
+    end
+    dummydf = similar(df[indexcols],0)
```

Comment:
`df[indexcols]` will copy and in general is going to be deprecated. If you want to use this design probably it is better to write `select(df, indexcols, copycols=false)`. Also only now you will catch a problem if you have duplicates in `indexcols` maybe it is better to handle it earlier.